scripts/metrics/ctest_metrics_xml_to_csv.py

```diff
@@ -0,0 +1,186 @@
+"""
+Copyright (c) Contributors to the Open 3D Engine Project.
+For complete copyright and license terms please see the LICENSE at the root of this distribution.
+
+SPDX-License-Identifier: Apache-2.0 OR MIT
+
+Scrapes metrics from CTest xml files and creates csv formatted files.
+"""
+import os.path
+import sys
+import csv
+import argparse
+import xml.etree.ElementTree as xmlElementTree
+import datetime
+
+from common import logging
+
+DEFAULT_CTEST_LOG_FILENAME = 'Test.xml'
+TAG_FILE = 'TAG'
+TESTING_DIR = 'Testing'
+
+def now_wrap():
+    """
+    Wrapper function for now() for unit testing purposes.
+    @return: datetime.now()
+    """
+    return datetime.datetime.now()
+
+
+def _get_default_csv_filename():
+    # Format default file name based off of date
+    now = now_wrap()
+    return f"{now.year}_{now.month:02d}_{now.day:02d}.csv"
+
+
+# Setup logging.
+logger = logging.get_logger("test_metrics")
+logging.setup_logger(logger)
+
+# Create the csv field header
+CTEST_FIELDS_HEADER = [
+    'test_name',
+    'status',
+    'duration_seconds'
+]
+
+
+def main():
+    # Parse args
+    args = parse_args()
+
+    # Construct the full path to the xml file
+    file_path = _get_test_xml_path(args.build_folder, args.ctest_log)
+
+    # Create csv file
+    if os.path.exists(args.csv_file):
+        logger.warning(f"The file {args.csv_file} already exists. It will be overriden.")
+    with open(args.csv_file, 'w', encoding='UTF8', newline='') as csv_file:
+        writer = csv.DictWriter(csv_file, fieldnames=CTEST_FIELDS_HEADER, restval='N/A')
+        writer.writeheader()
+
+        # Parse CTest xml and write to csv file
+        parse_ctest_xml_to_csv(file_path, writer)
+
+
+def parse_args():
+    parser = argparse.ArgumentParser(
+        description='This script assumes that a CTest xml file has been produced via the -T Test CTest option. The file'
+                    'should exist inside of the build directory. The xml file will be parsed and write to a csv file.')
+    parser.add_argument(
+        'build_folder',
```

Comment:
Argument appears [required](https://docs.python.org/3/library/argparse.html#required) for the script to function

Reply:
A positional arg is already required by default